vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AdamDeviceProvider.kt

```diff
@@ -51,6 +52,7 @@ class AdamDeviceProvider(
 
     private lateinit var client: AndroidDebugBridgeClient
     private lateinit var deviceEventsChannel: ReceiveChannel<List<Device>>
+    private val deviceEventsChannelLock = Object()
```

Comment:
Do we need java synchronized here or https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.sync/-mutex/index.html makes more sense?

Reply:
Fixed